src/server/plugins/engine/pageControllers/SummaryPageController.ts

```diff
@@ -96,45 +95,15 @@ export class SummaryPageController extends PageController {
       )
 
       /**
-       * iterates through the errors. If there are errors, a user will be redirected to the page
-       * with the error with returnUrl=`/${model.basePath}/summary` in the URL query parameter.
+       * Redirect back to pages with field errors
        */
       if (viewModel.errors) {
-        const errorToFix = viewModel.errors[0]
-        const { path: parts } = errorToFix
-        const section = parts[0]
-        const property = parts.length > 1 ? parts[parts.length - 1] : null
-        const pageWithError = model.pages.find((page) => {
-          if (page.section && page.section.name === section) {
-            let propertyMatches = true
-            let conditionMatches = true
-            if (property) {
-              propertyMatches =
-                page.collection.fields.filter(
-                  (component) => component.name === property
-                ).length > 0
-            }
-            if (
-              propertyMatches &&
-              page.condition &&
-              model.conditions[page.condition]
-            ) {
-              conditionMatches =
-                model.conditions[page.condition]?.fn(state) ?? false
-            }
-            return propertyMatches && conditionMatches
-          }
-          return false
-        })
-        if (pageWithError) {
-          const params = {
-            returnUrl: redirectUrl(`/${model.basePath}/summary`)
-          }
-          return redirectTo(
-            h,
-            `/${model.basePath}${pageWithError.path}`,
-            params
-          )
+        const errorField = viewModel.details
+          .flatMap(({ items }) => items.find(({ error }) => error) ?? [])
+          .at(0)
+
+        if (errorField) {
```

Comment:
@colinrotherham If it defaults to an empty array (which is truthy) this block will fire. Obv you know this lol, just confirming that's the intended behaviour?

Reply:
Ah I panicked then, there's a little `.at(0)` at the end to grab the first item